src/components/Table/index.js

```diff
@@ -0,0 +1,131 @@
+
+import HeaderCell from './components/TableHeaderCell';
+import RowCell from './components/TableRowCell';
+import { StyledTable } from './styles';
+
+const Table = React.forwardRef(
+  (
+    {
+      className,
+      columns,
+      data,
+      isSortable,
+      manualSorting,
+      onChangeSort,
+      initialState,
+    },
+    ref,
+  ) => {
+    const {
+      headerGroups,
+      rows,
+      prepareRow,
+      state: { sortBy },
+    } = useTable(
+      {
+        columns,
+        data,
+        manualSorting,
+        initialState,
+      },
+      useSortBy,
+    );
+
+    React.useMemo(
+      () =>
+        rows.forEach(row => {
+          prepareRow(row);
+        }),
+      [prepareRow, rows],
+    );
+
+    React.useEffect(() => {
+      onChangeSort(sortBy);
+    }, [onChangeSort, sortBy]);
+
+    return (
+      <StyledTable
+        className={cx('f-Table', className)}
+        cellPadding={0}
+        cellSpacing={0}
+        ref={ref}
+      >
+        <thead>
+          {headerGroups.map(({ getHeaderGroupProps, headers }) => (
+            <tr className={'f-Table-HeaderGroup'} {...getHeaderGroupProps()}>
+              {headers.map(
+                ({
+                  getHeaderProps,
+                  getSortByToggleProps,
+                  isSorted,
+                  isSortedDesc,
+                  render,
+                }) => (
+                  <HeaderCell
+                    {...getHeaderProps(isSortable && getSortByToggleProps())}
+                    isSorted={isSorted}
+                    isSortedDesc={isSortedDesc}
+                    isSortable={isSortable}
+                  >
+                    {render('Header')}
+                  </HeaderCell>
+                ),
+              )}
+            </tr>
+          ))}
+        </thead>
+
+        <tbody>
+          {rows.map(row => (
+            <tr className={'f-Table-RowGroup'} {...row.getRowProps()}>
+              {row.cells.map(({ getCellProps, render }) => (
+                <RowCell {...getCellProps()}>{render('Cell')}</RowCell>
+              ))}
+            </tr>
+          ))}
+        </tbody>
+      </StyledTable>
+    );
+  },
+);
+
+Table.displayName = 'Table';
+
+Table.propTypes = {
+  className: PropTypes.string,
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      Header: PropTypes.string,
+      accessor: PropTypes.string.isRequired,
+    }),
+  ),
+  data: PropTypes.arrayOf(PropTypes.shape({})),
+  isSortable: PropTypes.bool,
+  manualSorting: PropTypes.bool,
+  onChangeSort: PropTypes.func,
+  initialState: PropTypes.shape({
+    sortBy: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        desc: PropTypes.bool,
+      }),
+    ),
+  }),
+};
+
+Table.defaultProps = {
+  columns: [],
+  data: [],
+  isSortable: false,
+  onChangeSort: () => {},
+  manualSorting: false,
+};
+
+Table.HeaderCell = HeaderCell;
+Table.RowCell = RowCell;
```

Comment:
Originally we were exporting 

```
Table.HeaderGroup = HeaderGroup;
Table.HeaderCell = HeaderCell;
Table.RowGroup = RowGroup;
Table.RowCell = RowCell;
```

But both `HeaderGroup` and `RowGroup` are just an HTML element receiving ref, classname, props, children. All of what is done automatically by an element.

I didn't see the reason in exporting them either, since they don't have any functionality. 

Just want to confirm you're not using it anywhere in your apps for some reason @ugogo

Reply:
I think it's if we want to render a table piece by piece, not sure if that's used though